test/odbc/psqlodbc/test_pg_datatype_bigint.cpp

```diff
@@ -0,0 +1,1016 @@
+  odbcHandler.CloseStmt();
+
+  // Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(VIEW_NAME, {"*"}, vector<string>{COL1_NAME}));
+
+  // Make sure inserted values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(BIND_COLUMNS));
+
+  for (int i = 0; i < NUM_OF_INSERTS; i++) {
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+    ASSERT_EQ(pk_len, PK_BYTES_EXPECTED);
+    ASSERT_EQ(pk, i);
+
+    if (INSERTED_VALUES[i] != "NULL") {
+      ASSERT_EQ(data_len, DATA_BYTES_EXPECTED);
+      ASSERT_EQ(data, StringToBigInt(INSERTED_VALUES[i]));
+    }
+    else {
+      ASSERT_EQ(data_len, SQL_NULL_DATA);
+    }      
+  }
+
+  // Assert that there is no more data
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("VIEW", VIEW_NAME));
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_BigInt, Table_Unique_Constraints) {
+  const vector<pair<string, string>> TABLE_COLUMNS = {
+    {COL1_NAME, "INT PRIMARY KEY"},
+    {COL2_NAME, DATATYPE_NAME + " UNIQUE"}
+  };
+  const string UNIQUE_COLUMN_NAME = COL2_NAME;
+
+  const int PK_BYTES_EXPECTED = 4;
+  const int DATA_BYTES_EXPECTED = 8;
+
+  int pk;
+  long long int data;
+  SQLLEN pk_len;
+  SQLLEN data_len;
+  SQLLEN affected_rows;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler(Drivers::GetDriver(ServerType::PSQL));
+
+  const vector<string> INSERTED_VALUES = {
+    "9223372036854775807",
+    "123456789"
+  };
+  const int NUM_OF_INSERTED = INSERTED_VALUES.size();
+
+  const vector<tuple<int, int, SQLPOINTER, int, SQLLEN *>> BIND_COLUMNS = {
+    {1, SQL_C_LONG, &pk, 0, &pk_len},
+    {2, SQL_C_SBIGINT, &data, 0, &data_len}
+  };
+
+  string insert_string{};
+  string comma{};
+
+  for (int i = 0; i < NUM_OF_INSERTED; i++) {
+    insert_string += comma + "(" + std::to_string(i) + "," + INSERTED_VALUES[i] + ")";
+    comma = ",";
+  }
+
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS));
+  odbcHandler.CloseStmt();
+
+  // Check if unique constraint still matches after creation
+  char column_name[BUFFER_SIZE];
+  char type_name[BUFFER_SIZE];
+
+  const vector<tuple<int, int, SQLPOINTER, int>> TABLE_BIND_COLUMNS = {
+    {1, SQL_C_CHAR, column_name, BUFFER_SIZE},
+  };
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(TABLE_BIND_COLUMNS));
+
+  const string PK_QUERY =
+    "SELECT C.COLUMN_NAME FROM "
+    "INFORMATION_SCHEMA.TABLE_CONSTRAINTS T "
+    "JOIN INFORMATION_SCHEMA.CONSTRAINT_COLUMN_USAGE C "
+    "ON C.CONSTRAINT_NAME=T.CONSTRAINT_NAME "
+    "WHERE "
+    "C.TABLE_NAME='" +
+    TABLE_NAME.substr(TABLE_NAME.find('.') + 1, TABLE_NAME.length()) + "' "
+    "AND T.CONSTRAINT_TYPE='UNIQUE'";
```

Comment:
What is this trying to do? It's called PK_QUERY, but is it trying to get the unique key? Also, can't we use pg_constraint for this instead? Would be better than going through information_schema since it could be OID-based.